프로젝트 생성

noah0316 opened the pull request:
- closed: #3 #4 #1 #2 

## 🤝 그라운드 룰 논의 및 정의

@harimrim 과 함께 의논하여 그라운드 룰을 논의하고 [위키](https://github.com/codesquad-members-2023/swift-onban-05/wiki)에 정리하였습니다.

정의한 그라운드 룰은 아래와 같습니다.
- [코드 스타일 가이드](https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%91%8B-%EC%BD%94%EB%93%9C-%EC%8A%A4%ED%83%80%EC%9D%BC-%EA%B0%80%EC%9D%B4%EB%93%9C)
- [커밋 메시지 가이드](https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%A7%A9-%EC%BB%A4%EB%B0%8B-%EB%A9%94%EC%8B%9C%EC%A7%80-%EA%B0%80%EC%9D%B4%EB%93%9C)
- [프로젝트 구조 가이드](https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%97%82%EF%B8%8F-%ED%8F%B4%EB%8D%94-%EA%B5%AC%EC%A1%B0-%EA%B0%80%EC%9D%B4%EB%93%9C)

## ✍️ 학습계획, 구현계획 작성
- 학습계획과 구현계획을 GitHub의 Projects 기능을 이용하여 정리하였습니다.
- [링크](https://github.com/orgs/codesquad-members-2023/projects/6/views/1)

## 👷‍♂️ 프로젝트 생성

### 1️⃣ 의존성 관리도구 설치
- 의존성 관리도구로 요구사항에 맞게 CocoaPods를 사용하였습니다.
 - Toast를 표시하기 위해 [Toaster](https://github.com/devxoul/Toaster)를 설치하였습니다.
 - 정의된 룰대로 코드를 작성하기 위해 [SwiftLint](https://github.com/realm/SwiftLint)를 설치하였습니다.
 - [SwiftLint rule](https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%91%8B-%EC%BD%94%EB%93%9C-%EC%8A%A4%ED%83%80%EC%9D%BC-%EA%B0%80%EC%9D%B4%EB%93%9C#%EF%B8%8F-%EC%BD%94%EB%93%9C-%EC%8A%A4%ED%83%80%EC%9D%BC-%EA%B0%80%EC%9D%B4%EB%93%9C%EB%A5%BC-%EA%B3%A0%EB%A0%A4%ED%95%98%EC%97%AC-%EC%84%A4%EC%A0%95%ED%95%9C-swiftlintyml)
 - Pods 폴더를 불필요하게 tracking하게 하지 않기 위해 .gitignore에 추가하였습니다.

### 2️⃣ 프로젝트 구조 설정
@harimrim 과 함께 논의하여 정의한 [프로젝트 구조가이드](https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%97%82%EF%B8%8F-%ED%94%84%EB%A1%9C%EC%A0%9D%ED%8A%B8-%EA%B5%AC%EC%A1%B0-%EA%B0%80%EC%9D%B4%EB%93%9C)에 맞추어 초기 프로젝트 구조를 설정하였습니다.

noah0316 committed:
온반-iOS-05 폴더 구조 가이드를 적용한 프로젝트 생성

- 페어(@harimrim)와 함께 논의를 통해 설정한 온반의 폴더 구조 가이드를 적용하여 프로젝트를 생성하였습니다.
- 폴더 구조 가이드 링크 : https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%97%82%EF%B8%8F-%ED%8F%B4%EB%8D%94-%EA%B5%AC%EC%A1%B0-%EA%B0%80%EC%9D%B4%EB%93%9C

harimrim committed:
의존성 관리 파일 추가

- 의존성 관리를 위한 cocoapod 파일 추가

harimrim committed:
SwiftLint 라이브러리 기입

- 코드 스타일을 맞추기 위한 SwiftLint 라이브러리 기입

harimrim committed:
Pods 폴더 트랙킹 제외

- pod install 시 Pods 폴더는 생성되기 때문에 불필요한 폴더를 추적하지 않기 위해 Pods 파일 주석 제거

harimrim committed:
SwiftLint 라이브러리 설치

- Podfile을 이용하여 SwiftLint 설치

harimrim committed:
Toaster 라이브러리 기입

- 섹션 제목 영역을 터치하면 섹션 제목과 함께 상품 개수를 toast형태로 표시하기 위한 라이브러리 기입

harimrim committed:
Toaster 라이브러리 설치

- Podfile을 이용하여 Toaster 설치

noah0316 committed:
.DS_Store 파일 tracking 제외

- 프로젝트와 무관한 .DS_Store 파일을 tracking 하지않기 위해 .DS_Store 파일 tracking 제외

noah0316 committed:
SwiftLint 룰 생성

- 온반-iOS-05 코드 스타일 가이드를 적용한 SwiftLint 룰 생성
- 링크: https://github.com/codesquad-members-2023/swift-onban-05/wiki/%F0%9F%91%8B-%EC%BD%94%EB%93%9C-%EC%8A%A4%ED%83%80%EC%9D%BC-%EA%B0%80%EC%9D%B4%EB%93%9C

noah0316 committed:
Deployment target 설정

- 최소 deployment target 16.0으로 설정

noah0316 committed:
불필요한 공백 제거

- 설정한 SwiftLint rule에 맞추어 불필요한 공백 제거

godrm commented:
확인했습니다
프로젝트 저장소를 다룰 때는 README.md 파일에 위에 정리한 내용들을 포함해도 좋을꺼 같습니다
머지하고 진행하셔도 됩니다.

noah0316 committed:
진행사항 업데이트

- 그라운드 룰 논의 및 정의와 관련된 진행사항을 반영하였습니다.
- 학습계획, 구현계획과 관련된 진행사항을 반영하였습니다.
- 프로젝트 생성과 관련된 진행사항을 반영하였습니다.

noah0316 commented:
> 확인했습니다 프로젝트 저장소를 다룰 때는 README.md 파일에 위에 정리한 내용들을 포함해도 좋을꺼 같습니다 머지하고 진행하셔도 됩니다.

확인하였습니다.

c22502dd1abac57dd99f09d6271e207b86928aed 에 업데이트 하였습니다!

noah0316 commented:
머지 하겠습니다 :)
